cmd/cstorvolume-claim/cvc/cstorvolumeclaim.go

```diff
@@ -0,0 +1,399 @@
+				IntVal: 7777,
+			},
+		},
+		corev1.ServicePort{
+			Name:     "mgmt",
+			Port:     6060,
+			Protocol: "TCP",
+			TargetPort: intstr.IntOrString{
+				IntVal: 6060,
+			},
+		},
+		corev1.ServicePort{
+			Name:     "exporter",
+			Port:     9500,
+			Protocol: "TCP",
+			TargetPort: intstr.IntOrString{
+				IntVal: 9500,
+			},
+		},
+	}
+)
+
+// getStorageClass return storageclass object for a given storageClass Name.
+// or error if any.
+func getStorageClass(storageClassName string,
+) (*storagev1.StorageClass, error) {
+	if storageClassName == "" {
+		return nil, errors.New("failed to get storageclass: storageclass missing")
+	}
+	scObj, err := sc.NewKubeClient().Get(storageClassName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"failed to get storageclass {%s}",
+			storageClassName,
+		)
+	}
+	return scObj, nil
+}
+
+// getReplicationFactor gets the Replication and Consistency factor from the
+// storageclass
+func getReplicationFactor(scName string,
+) (int, int, error) {
+
+	scObj, err := getStorageClass(scName)
+	if err != nil {
+		return 0, 0, errors.Wrapf(
+			err,
+			"failed to get storageclass obj {%s}",
+			scName,
+		)
+	}
+	count := scObj.Parameters["replicaCount"]
+
+	rfactor, err := strconv.Atoi(count)
+	if err != nil {
+		return 0, 0, errors.Wrapf(
+			err,
+			"failed to convert to int {%s}",
+			count,
+		)
+	}
+	return rfactor, (rfactor/2 + 1), nil
+
+}
+
+// getFromParameters gets replicaCount and storagePoolClaim from
+// storageclass parameter
+func getFromParameters(scName string,
+) (int, string, error) {
+
+	scObj, err := getStorageClass(scName)
+	if err != nil {
+		return 0, "", errors.Wrapf(
+			err,
+			"failed to get storageclass obj {%s}",
+			scName,
+		)
+	}
+	count := scObj.Parameters["replicaCount"]
+	rCount, err := strconv.Atoi(count)
+	if err != nil {
+		return 0, "", errors.Wrapf(
+			err,
+			"failed to convert to int {%s}",
+			count,
+		)
+	}
+
+	spcName := scObj.Parameters["storagePoolClaim"]
+	return rCount, spcName, nil
+}
+
+// listCStorPools get the list of available pool using the storagePoolClaim
+// as labelSelector.
+func listCStorPools(
+	spcName string,
+	replicaCount int,
+) (*apis.CStorPoolList, error) {
+
+	if spcName == "" {
+		return nil, errors.New("failed to list cstorpool: spc missing")
+	}
+
+	labelSelector := "openebs.io/storage-pool-claim=" + spcName
+
+	cstorPoolList, err := csp.KubeClient().List(metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"failed to list cstorpool for spc {%s}",
+			spcName,
+		)
+	}
+	if len(cstorPoolList.Items) < replicaCount {
+		return nil, errors.New("not enough pools available to create replicas")
+	}
+	return cstorPoolList, nil
+}
+
+// createTargetService creates cstor volume target service
+func createTargetService(storageClassName string,
+	claim *apis.CStorVolumeClaim,
+) (*corev1.Service, error) {
+
+	labels := map[string]string{
```

Comment:
match all the labels with current runtask.. I see the missing ones are: storage-engine-type, cas-type, version etc..
if you plan to remove anything (including the pvc details), please mention it explicitly.
Please check for other resources as well.